### PR TITLE
Block library: Refactor Verse block to use class names for text align

### LINKED ISSUE #15751: Verse Block: Alignment uses inline styles
Currently the verse block get an inline style after selecting a specific alignment.

`<pre style="text-align:left" class="wp-block-verse">Content</pre>`

This is an unexpected behavior and should work like this:

`<pre class="wp-block-verse alignleft">Content</pre>`

Without adding the alignment class, I can't set any specific theme side styles for the alignment of verse block.

> To style a poem block, the alignment class is necessary to get more design possibilities.

Is there a reason why you should not add the alignment class to this block type?

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Related: https://github.com/WordPress/gutenberg/issues/16027, https://github.com/WordPress/gutenberg/issues/15751

This PR follows-up #16035 and fixes #15751.

> Currently the verse block get an inline style after selecting a specific alignment.
> 
> `<pre style="text-align:left" class="wp-block-verse">Content</pre>`
> 
> This is an unexpected behavior and should work like this:
> 
> `<pre class="wp-block-verse alignleft">Content</pre>`
> 
> Without adding the alignment class, I can't set any specific theme side styles for the alignment of verse block.

Before:
```html
<!-- wp:verse {"textAlign":"right"} -->
<pre style="text-align:right" class="wp-block-verse">A <em>verse</em>…<br>And more!</pre>
<!-- /wp:core/verse -->
```

After:
```html
<!-- wp:verse {"textAlign":"right"} -->
<pre class="wp-block-verse has-text-align-right">A <em>verse</em>…<br>And more!</pre>
<!-- /wp:verse -->
```

## How has this been tested?

Using one of the existing branches (`master` probably) add a few Verse blocks and set different text alignments and save your post. Open the same post with this branch and ensure that it still works

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/block-library/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 ## Master (unreleased)
 
+### Enhancements
+
+- Heading block uses `has-text-align-*` class names rather than inline style for text alignment.
```
Comment (draganescu) [inline packages/block-library/CHANGELOG.md]: Why is this here?

**Review by draganescu — CHANGES_REQUESTED** [review]

File: packages/block-library/CHANGELOG.md (hunk 1 above)
Comment (gziolo) [reply packages/block-library/CHANGELOG.md]: It was missed in other PR, I didn't want to bother myself with another PR and going through the review process which takes ages.

**Review by draganescu — APPROVED** [review]

**Comment by gziolo** [comment]
Applying `Needs Dev Note` as this might have an impact on themes which used the presence of style as a way to change the visual aspect of the block.
